docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,1652 @@
+    // Can have at most 1 CIDR for each IP family (IPv4 and IPv6).
+    // Must not overlap with:
+    //  any of the pod subnets used by the selected networks.
+    //  any of the transit subnets used by the selected networks.
+    //  any of the service CIDR range used in the cluster.
+    //  any of the join subnet of the selected networks to be connected.
+    //  any of the masquerade subnet range used in the cluster.
+    //  any of the node subnets for choosen by the platform.
+    //  any of other connect subnets for other ClusterNetworkConnects that might be selecting same networks.
+    //
+    // Does not have a default value for the above reason so
+    // that user takes care in setting non-overlapping subnets.
+    // Cannot be mutated once set.
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:MinItems=1
+    // +kubebuilder:validation:MaxItems=2
+    // +kubebuilder:validation:XValidation:rule="self == oldSelf", message="connectSubnets is immutable"
+    // +required
+    // +kubebuilder:validation:XValidation:rule="size(self) != 2 || !isCIDR(self[0].cidr) || !isCIDR(self[1].cidr) || cidr(self[0].cidr).ip().family() != cidr(self[1].cidr).ip().family()", message="When 2 CIDRs are set, they must be from different IP families"
+    ConnectSubnets []ConnectSubnet `json:"connectSubnets"`
+
+    // connectivityEnabled specifies which connectivity types should be enabled for the connected networks.
+    //
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:MinItems=1
+    // +kubebuilder:validation:MaxItems=2
+    // +kubebuilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))",message="connectivityEnabled cannot contain duplicate values"
+    ConnectivityEnabled []ConnectivityType `json:"connectivityEnabled"`
+}
+
+// +kubebuilder:validation:XValidation:rule="isCIDR(self) && cidr(self) == cidr(self).masked()", message="CIDR must be a valid network address"
+// +kubebuilder:validation:MaxLength=43
+type CIDR string
+
+// +kubebuilder:validation:XValidation:rule="!has(self.layer3NetworkPrefix) || !isCIDR(self.cidr) || self.layer3NetworkPrefix > cidr(self.cidr).prefixLength()", message="layer3NetworkPrefix must be smaller than CIDR subnet"
+// +kubebuilder:validation:XValidation:rule="!has(self.layer3NetworkPrefix) || !isCIDR(self.cidr) || (cidr(self.cidr).ip().family() != 4 || self.layer3NetworkPrefix < 32)", message="layer3NetworkPrefix must < 32 for ipv4 CIDR"
+type ConnectSubnet struct {
+  // CIDR specifies ConnectSubnet, which is split into smaller subnets for every connected network.
+  // This CIDR should be containing 2*((Number of L3 networks*Max Number of Nodes)+Number of L2 networks) IPs.
+  //
+  // +required
+  CIDR CIDR `json:"cidr"`
+
+  // Layer3NetworkPrefix specifies the prefix length for every connected network.
+  // This prefix length should be smaller than the CIDR prefix length.
```

Comment:
This prefix length should be equal to or longer than the length of the CIDR prefix

Reply:
haha yea in my head I had it reversed - bigger number is smaller length ugh. let me correct it.